base.js

```diff
@@ -300,8 +300,16 @@ function getHash(url) {
 	return hash;
 }
 
-function parseValue(value, options) {
-	if (options.parseNumbers && !Number.isNaN(Number(value)) && (typeof value === 'string' && value.trim() !== '')) {
+function parseValue(value, options, type) {
+	if (type === 'string' && (typeof value === 'string')) {
+		return value;
+	}
+
+	if (typeof type === 'function' && (typeof value === 'string')) {
+		value = type(value);
+	} else if (type === 'number' && !Number.isNaN(Number(value)) && (typeof value === 'string' && value.trim() !== '')) {
```

Comment:
Do the `!Number.isNaN(Number(value))` check last as it's the most expensive one.